src/bots/ama/subCommands/listEvents.js

```diff
@@ -10,50 +11,78 @@ const handler = async () => {
   if (events.length === 0) {
     await client.message.channel.send(getStrings().noEvents);
   } else {
-    client.message.channel.send(formatEvents(events));
+    let parsedCmd = parseCommandString();
+
+    client.message.channel.send(
+      formatEvents(events, parsedCmd.arguments.length > 0)
+    );
   }
 };
 
 /**
  * Formats events into a readable string
  * @param {Array.<Object.<string, any>>} events - list of event objects
+ * @param {boolean} showIds - trigger that shows event Database IDs if set to true
  * @returns {string} - message string with all events
  */
-const formatEvents = (events) => {
+const formatEvents = (events, showIds) => {
   let currentDate = new Date();
   let splitDate = partition(events, (e) => e.date > currentDate);
+  let eventString = '';
+
+  if (splitDate[1].length > 0) {
+    let pastEvents = '';
+    let pastEventObjects = splitDate[1];
 
-  let upcomingEvent = getStrings(
-    `${splitDate[0][0].date.toDateString()}: ${splitDate[0][0].title}`
-  ).upcomingEvent;
+    for (let s in pastEventObjects) {
+      let next = pastEventObjects[s];
 
-  let futureEvents = '';
+      pastEvents += `${
+        showIds ? next.id + ' ' : ''
+      }${next.date.toDateString()}: ${next.title}\n`;
+    }
 
-  for (let i = 1; i < splitDate[0].length; i++) {
-    let next = splitDate[0][i];
+    pastEvents = getStrings(pastEvents).pastEvents;
 
-    futureEvents += `${next.date.toDateString()}: ${next.title}\n`;
+    eventString += pastEvents + '\n';
   }
 
-  futureEvents = getStrings(futureEvents).futureEvents;
+  if (splitDate[0].length > 0) {
+    let futureEventObjects = splitDate[0];
+    let upcomingEventObject = futureEventObjects[0];
 
-  let pastEvents = '';
+    let upcomingEvent = getStrings(
+      `${
+        showIds ? upcomingEventObject.id + ' ' : ''
+      }${upcomingEventObject.date.toDateString()}: ${upcomingEventObject.title}`
+    ).upcomingEvent;
 
-  for (let s in splitDate[1]) {
-    let next = splitDate[1][s];
+    eventString += upcomingEvent + '\n';
 
-    pastEvents += `${next.date.toDateString()}: ${next.title}\n`;
-  }
+    let futureEvents = '';
+
+    for (let i = 1; i < futureEventObjects.length; i++) {
+      let next = futureEventObjects[i];
```

Comment:
same as my pastEvent comment. also for for loops where you only need the value and not the index just do:
```suggestion
for (let futureEvent of futureEventObjects)
```

Reply:
you have this for some other for loops too, i won't comment there but would also be good to fix those

Reply:
can't use enhanced for loop here since I'm working off a subset (1..n) not 0..n.
But I'll try fix the others.